### PR TITLE
Build mint into a Docker image

### PR DESCRIPTION
**Pull request by tobias-mintlify**
# Summary

Builds `mint` into a Docker image for the CLI to consume. Most of the files are copied from the `local-preview` folder into a new package that gets copied into Docker.

# Test Plan

You can build with: `docker build -t mint .` while running Docker Desktop.

No production impact yet. The Docker image is not being used.

# Future Work

Mount the user directory into Docker.

Run listener and `yarn dev` in parallel.

Delete listener code from the CLI and run Docker instead.

### CODE REVIEW

File: DOCKER_README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+## Prerequisites
+
+Install docker: [https://docs.docker.com/desktop/install/mac-install/](https://docs.docker.com/desktop/install/mac-install/)
+
+## Building the Mint image
+
+Building the docker image:
+
+`docker build -t mint .`
```
Comment (hahnbeelee) [inline DOCKER_README.md:13]: Where do I run this? In my terminal or inside Docker Desktop? After I downloaded Docker Desktop and tried running this in my terminal I got the error message "zsh: command not found: docker". I know this all seems trivial but coming from no Docker experience I'd appreciate more instruction :bongo-pray:

**Review by hahnbeelee — CHANGES_REQUESTED** [review]

File: DOCKER_README.md (hunk 1 above)
Comment (tobias-mintlify) [reply DOCKER_README.md:13]: Ah yea! You have to run this in the terminal while Docker Desktop is open.

Comment (hahnbeelee) [reply DOCKER_README.md:13]: Note I downloaded docker desktop and then I had to close my terminal in order to access the docker command. Might be helpful to note that for future devs.

Comment (hahnbeelee) [reply DOCKER_README.md:13]: Is the only step ensuring `docker build -t mint .` works? Are there other things I should test?

**Comment by hahnbeelee** [comment]
Cool I see my mint image! (why is it 5GB 👁️ )
image

**Comment by tobias-mintlify** [comment]
Node's Docker image is roughly 1Gb. Most of the rest comes from dependencies.

**Review by hahnbeelee — APPROVED** [review]
wasn't able to review in depth but I am in a rush to get this out the door :sad-bufo-vibe: @tobias-mintlify I might make comments in the future about things on this PR post-merge.

**Pull request merged**